sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c

```diff
@@ -0,0 +1,288 @@
+            iface->rtr_sol_count++;
+        }
+
+        DEBUG("6lo nd: retransmit multicast rtr sol in %" PRIu32 " sec\n", interval);
+        iface->rtr_sol_count--;
+    }
+    _rtr_sol_reschedule(iface, interval);
+    mutex_unlock(&iface->mutex);
+    gnrc_ndp_internal_send_rtr_sol(iface->pid, NULL);
+}
+
+void gnrc_sixlowpan_nd_uc_rtr_sol(gnrc_ipv6_nc_t *nce)
+{
+    assert(gnrc_ipv6_netif_get(nce->iface)->flags & GNRC_IPV6_NETIF_FLAGS_SIXLOWPAN);
+    /* neighbor is not a router anymore */
+    if (!(nce->flags & GNRC_IPV6_NC_IS_ROUTER) || ipv6_addr_is_unspecified(&nce->ipv6_addr)) {
+        /* and there are no routers anymore */
+        if (gnrc_ipv6_nc_get_next_router(NULL) == NULL) {
+            /* start search for routers */
+            gnrc_sixlowpan_nd_init(gnrc_ipv6_netif_get(nce->iface));
+        }
+        /* otherwise ignore this call */
+        return;
+    }
+    /* next RS is rescheduled by RA handle function */
+    gnrc_ndp_internal_send_rtr_sol(nce->iface, &nce->ipv6_addr);
+}
+
+kernel_pid_t gnrc_sixlowpan_nd_next_hop_l2addr(uint8_t *l2addr, uint8_t *l2addr_len,
+                                               kernel_pid_t iface, ipv6_addr_t *dst)
+{
+    ipv6_addr_t *next_hop = NULL;
+    gnrc_ipv6_nc_t *nc_entry = NULL;
+
+#ifdef MODULE_GNRC_IPV6_EXT_RH
+    ipv6_hdr_t *hdr;
+    gnrc_pktsnip_t *ipv6;
+    LL_SEARCH_SCALAR(pkt, ipv6, type, GNRC_NETTYPE_IPV6);
+    assert(ipv6);
+    hdr = ipv6->data;
+    next_hop = ipv6_ext_rh_next_hop(hdr);
+#endif
+#ifdef MODULE_FIB
+    ipv6_addr_t next_hop_actual;    /* FIB copies address into this variable */
+    /* don't look-up link local addresses in FIB */
+    if ((next_hop == NULL) && !ipv6_addr_is_link_local(dst)) {
+        size_t next_hop_size = sizeof(ipv6_addr_t);
+        uint32_t next_hop_flags = 0;
+        if ((next_hop == NULL) &&
+            (fib_get_next_hop(&gnrc_ipv6_fib_table, &iface, next_hop_actual.u8, &next_hop_size,
+                              &next_hop_flags, (uint8_t *)dst,
+                              sizeof(ipv6_addr_t), 0) >= 0) &&
+            (next_hop_size == sizeof(ipv6_addr_t))) {
+            next_hop = &next_hop_actual;
+        }
+    }
+#endif
+    /* next hop determination according to: https://tools.ietf.org/html/rfc6775#section-5.6 */
+    if ((next_hop == NULL) && ipv6_addr_is_link_local(dst)) {   /* prefix is "on-link" */
+        /* multicast is not handled here anyway so we don't need to check that */
+        next_hop = dst;
+    }
+    else if (next_hop == NULL) {                                /* prefix is off-link */
+        next_hop = gnrc_ndp_internal_default_router();
+    }
+
+    /* address resolution of next_hop: https://tools.ietf.org/html/rfc6775#section-5.7 */
+    if (ipv6_addr_is_link_local(next_hop)) {
+        kernel_pid_t ifs[GNRC_NETIF_NUMOF];
+        size_t ifnum = gnrc_netif_get(ifs);
+        assert(ifnum == 1); /* 6LoWPAN hosts should only have one IEEE 802.15.4 interface */
```

Comment:
Yes, but that has nothing to do with ND, because ND is for IP. I now take the first 6LoWPAN interface if there is more than one. Is there is only one, I assume it to be a 6LoWPAN interface because otherwise we wouldn't be in this function. Please check the updated code.